### PR TITLE
return errors where appropriate

### PR DESCRIPTION
**Pull request by perbu**
Currently the package returns a boolean indicating success. 

I've also done some minor tweaks for readability and adhering to conventions.

### CODE REVIEW

**Comment by dmke** [comment]
I've changed base from `master` to `v2`. I try to look into this further later (maybe tomorrow).

**Comment by perbu** [comment]
(sorry, I didn't see you changed the PR, so I was opening a new one when I saw what you did)

**Comment by dmke** [comment]
Don't worry, the Github UI can be confusing at times...

(Looks like I need to update the CI tooling. The test matrix still targets Go 1.13 :eyes:)

File: items.go
Comment (dmke) [inline items.go]: Global search+replace is seldomly a good idea :)

This should still read

```suggestion
	itemError itemType = iota // error occurred; item.val is error message
```

or similar.

Comment (perbu) [reply items.go]: Oh. This was Goland doing refactoring (which is kinda like a search/replace). Mea culpa.

**Comment by perbu** [comment]
oh. there are errstr all over the place now. I'll fix.

File: lexer.go
Comment (dmke) [inline lexer.go]: Missed one :)

File: test_test.go
Comment (dmke) [inline test_test.go]: And another.

**Review by dmke — CHANGES_REQUESTED** [review]
Looks good so far. I'll merge this soon, after I've reconfigured the CI pipeline.

**Comment by perbu** [comment]
Thanks for looking at this. Sorry for the errstr stuff. 

The question is what we're suppose to do with the deprecated stuff. I can have a stab a clearing it out once this PR is merged, or perhaps you wanna do it? Let me know and I can plan accordingly.

**Comment by dmke** [comment]
> Sorry for the errstr stuff.

Don't worry, things happen.

> The question is what we're suppose to do with the deprecated stuff. I can have a stab a clearing it out once this PR is merged, or perhaps you wanna do it? Let me know and I can plan accordingly.

Feel free to clear them out, while you're at it :)

I've pushed a few commits to update dependencies and the CI workflows, so you may want to rebase your PR on top of them.

**Comment by perbu** [comment]
Removed the deprecated calls as requested. Gone over the existing todos and resolved as many as I could. There is one left, which is wrt to a potential error which might show up during handling anonymous sections, something I know know nothing about.

I've been unable to figure out how to get switch my fork to use the v2 branch directly, so I haven't been able to rebase. Hope it doesn't cause much problems.

I'm quite happy with the state of it as it is now. Lemme know.

**Comment by dmke** [comment]
> I've been unable to figure out how to get switch my fork to use the v2 branch directly

Assuming `git remote -v` displays something like

```
origin	git@github.com:perbu/go-uci (fetch)
origin	git@github.com:perbu/go-uci (push)
upstream	git@github.com:digineo/go-uci.git (fetch)
upstream	git@github.com:digineo/go-uci.git (push)
```

i.e. `upstream` points to this repo, and `origin` to yours, to rebase you'll need to:

1. Fetch my `v2` branch
   
   ```console
   $ git fetch upstream 
   ```

2. Move your commits onto that branch:

   ```console
   $ git rebase upstream/master
   ```

   Note: This may or may not) create a merge conflict you'd need to resolve before continuing (I've fixed some linter issues, which you might have as well - if we've applied the same solution, git won't complain). Should things go south, you can abort the rebase attempt with `git rebase --abort`, even after solving one or more conflicts.

3. Push your changes (forcefully):

   ```console
   $ git push --force-with-lease origin master
   ```

Would you try that? If not, or if problems arise, let me know, I should be able to do this on my end.

**Comment by perbu** [comment]
I think I managed to pull it off.

**Comment by dmke** [comment]
Looks good. I'll merge this when I'm back home.

**Review by dmke — APPROVED** [review]

**Comment by dmke** [comment]
I've squashed your commits into 60c1481, tagged as `v2.0-rc.1` and cleaned some minor things up. The PR somehow contained a merge commit with master, which led to the commits being duplicated... Not sure how this could happen :)

There still was a `(*tree).Set` method lying around, but `Set` was already removed from the `Tree` interface. I'm now considering renaming `SetType` to `Set`, since one argument already contains the `Type` keyword:

```go
uci.SetType("pkg", "sect", uci.TypeOption, "val")
// vs.
uci.Set("pkg", "sect", uci.TypeOption, "val")
```

**Pull request closed without merging**
